src/main/java/uk/gov/hmcts/reform/ccd/document/am/controller/endpoints/CaseDocumentAm.java

```diff
@@ -252,4 +252,28 @@
         @ApiParam(value = "Comma-separated list of roles of the currently authenticated user. If provided will be used for authorisation.")
         @RequestHeader(value = "user-roles", required = false) String userRoles);
 
+
+    @ApiOperation(value = "Retrieves the hashcode for document Id", nickname = "generateHashCode",
+        notes = "This API will return the hashed token required for document upload functionality.",
+        response = String.class, tags = {TAG})
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Success", response = StoredDocumentHalResource.class),
+        @ApiResponse(code = 401, message = UNAUTHORIZED, response = String.class),
+        @ApiResponse(code = 403, message = FORBIDDEN, response = ErrorMap.class, responseContainer = "List"),
+        @ApiResponse(code = 400, message = "Bad Request", response = String.class)})
+    @RequestMapping(value = "/cases/documents/{documentId}/token", produces = {APPLICATION_JSON}, method = RequestMethod.GET)
+    ResponseEntity<Object> generateHashCode(
+
+        @ApiParam(value = S2S_API_PARAM, required = true)
+        @RequestHeader(value = SERVICE_AUTHORIZATION, required = true) String serviceAuthorization,
```

Comment:
I think, we have missed the authorisation token which is mandatory for all APIs. could you please add that.

Reply:
Changes are implemented.